### PR TITLE
Implemented 1-bit adam

### LINKED ISSUE #69: Implement 1-Bit Adam
Integrate 1-bit Adam into our model. The DeepSpeed tutorial can be found [here](https://www.deepspeed.ai/tutorials/onebit-adam/)

### PR DESCRIPTION
**Pull request by StellaAthena**
Closes #69 

There are some problems with the `gpt3small` training script that snuck in at some point. I've pushed some changes that improve this but there's more work to do and it doesn't yet run. See #71

### CODE REVIEW

File: configs/deepspeed_zero1.json [hunk 1]
```diff
@@ -8,9 +8,11 @@
       "job_name": "gptneox"
     },
     "optimizer": {
-      "type": "Adam",
+      "type": "OneBitAdam",
       "params": {
-        "lr": 0.00015
+        "lr": 2e-4,
```
Comment (leogao2) [inline configs/deepspeed_zero1.json:13]: changed lr?

File: configs/deepspeed_zero2.json [hunk 2]
```diff
@@ -8,10 +8,11 @@
       "job_name": "gptneox"
     },
     "optimizer": {
-      "type": "Adam",
+      "type": "OneBitAdam",
       "params": {
-        "lr": 0.00015
-      }
+        "lr": 2e-4,
```
Comment (leogao2) [inline configs/deepspeed_zero2.json:13]: changed lr?

**Review by leogao2 — APPROVED** [review]

File: configs/deepspeed_zero1.json (hunk 1 above)
Comment (StellaAthena) [reply configs/deepspeed_zero1.json:13]: Recommended by DeepSpeed for 1-Bit Adam

File: configs/deepspeed_zero2.json (hunk 2 above)
Comment (StellaAthena) [reply configs/deepspeed_zero2.json:13]: Recommended by DeepSpeed for 1-Bit Adam

**Pull request merged**